src/loan_form/LoanForm.jsx

```diff
@@ -4,21 +4,59 @@ import LoanEntries from './LoanEntries'
 import AddLoan from './buttons/AddLoanButton'
 import Calculate from './buttons/CalculateButton'
 import emptyLoan from '../models/loan/emptyLoan'
-
-const calculate = e => console.log('calculate')
+import Calculation from './loan_calculate/Calculation'
 
 export default class LoanForm extends React.Component {
   constructor(props) {
     super(props)
     this.state = {
-      loans: [emptyLoan(0)]
+      loans: [emptyLoan(0)],
+      maxTotalMonths: 0
     }
+    this.handleChange = this.handleChange.bind(this)
   }
 
   createLoan() {
     const lastId = this.state.loans[this.state.loans.length - 1].id
     return emptyLoan(lastId + 1)
   }
+  
+  @autobind
+  handleChange(id, name, value) {
+    const newLoans = this.state.loans
+
+    for (var i = 0; i < this.state.loans.length; i++) {
+      if (this.state.loans[i]['id'] == parseInt(id)) {
+        switch(name) {
+          case 'title':
+            newLoans[i].title = value
+            break
+          case 'amount-owed':
+            newLoans[i].amountOwed = value
+            newLoans[i].totalMonths = this.findLoanLength(parseFloat(newLoans[i].amountOwed),
+                                                          parseFloat(newLoans[i].interestRate),
+                                                          parseFloat(newLoans[i].monthlyPayment))
+            console.log(newLoans[i])
+            break
+          case 'interest-rate':
+            newLoans[i].interestRate = value
+            newLoans[i].totalMonths = this.findLoanLength(parseFloat(newLoans[i].amountOwed),
+                                                          parseFloat(newLoans[i].interestRate),
+                                                          parseFloat(newLoans[i].monthlyPayment))
+            break
+          case 'monthly-payment':
+            newLoans[i].monthlyPayment = value
+            newLoans[i].totalMonths = this.findLoanLength(parseFloat(newLoans[i].amountOwed),
+                                                          parseFloat(newLoans[i].interestRate),
+                                                          parseFloat(newLoans[i].monthlyPayment))
+            break
```

Comment:
Okay, so given that in js you can access members of an object via `object.member` _or_ `object['member']` and given that the only real difference in each of these cases is which field on `newLoan` is set to `value`, i think you could write yourself a couple of helper methods and use those here. One method could be a method that accepts the loan object, the name of the field, and the new value and sets it accordingly (this is basically really simple "ad-hoc polymorphism"). The other method could be one that accepts the loan object and updates it's `totalMonths` field.

Another way to go with this would be to add a method to the loan object itself that can calculate `totalMonths` using it's slots.

Reply:
good call